gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -96,12 +97,21 @@ private File downloadOfflineBaseMapSource(OfflineBaseMapSource offlineBaseMapSou
 
   /** Enqueue a single area and its tile sources for download. */
   private Completable enqueueDownload(OfflineArea area, ImmutableList<TileSource> tileSources) {
+
     return localDataStore
         .insertOrUpdateOfflineArea(area.toBuilder().setState(State.IN_PROGRESS).build())
         .andThen(
             Completable.merge(
                 stream(tileSources.asList())
-                    .map(localDataStore::insertOrUpdateTileSource)
+                    .map(
```

Comment:
My intuition is that there's a simpler way of doing this using RxJava fromIterable and recombining the result (`combine()`?) but would take some tinkering to figure it out. If you don't want to explore that now please add a TODO to simplifying this block.

Reply:
b86a87a